tests/Algorithms/Ed25519ToX25519Test.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Linq;
+using NSec.Cryptography;
+using Xunit;
+
+namespace NSec.Tests.Algorithms
+{
+    public class Ed25519ToX25519Test
+    {
+        [Fact]
+        public void TestEd25519_to_X25519()
+        {
+            var ed25519 = Key.Create(SignatureAlgorithm.Ed25519,
+                new KeyCreationParameters() {ExportPolicy = KeyExportPolicies.AllowPlaintextExport});
+            var x25519pub = Ed25519ToX25519.SignEd25519PublicToCurve25519(ed25519.PublicKey);
+            var x25519priv = Ed25519ToX25519.SignEd25519SecretToCurve25519(ed25519);
+            Assert.Equal(x25519priv.PublicKey, x25519pub);
```

Comment:
Does this check that the key here is actually correct (ie, key generated corresponds with the ed25519 key) or that the key is simply valid (ie, public key returned corresponds with the private key returned)?

Reply:
There are basically two paths to get a converted public key:

```
Ed25519      SignEd25519SecretToCurve25519      X25519
private   ----------------------------------->  private
  key                                             key

   |                                              |
   | .PublicKey                        .PublicKey |
   |                                              |
   v                                              v

Ed25519      SignEd25519PublicToCurve25519      X25519
public    ----------------------------------->  public
  key                                             key
```

This test just checks that the converted public keys for both of the paths are actually the same (which is a good thing to test!).

It doesn't test whether the conversion was correct, though. So the next step would be to add some tests with verified test vectors to ensure that everything is right.

Reply:
Thanks!